test/engine/null.test.lua

```diff
@@ -612,3 +612,14 @@ s:insert{1, -1, 1}
 sk2 = s:create_index('sk2', {parts={{2, 'number', is_nullable=true}}})
 s:insert{2, nil, 2} --error
 s:drop()
+
+-- gh-5674: ignoring index part options when type is omitted
+s = box.schema.space.create('test', {engine=engine})
+_ = s:create_index('pk', {parts = {1, 'int'}})
+sk = s:create_index('sk', {parts = {2, is_nullable=true}})
+sk.parts
+sk1 = s:create_index('sk1', {parts = {2, is_nullable=true, collation='unicode'}})
+sk1.parts
+sk2 = s:create_index('sk2', {parts = {2, type='string', is_nullable=true, collation='unicode'}})
```

Comment:
Just in case I'd add test with several key parts (both in single array and separate array per key part).

Reply:
Otherwise LGTM